programs/uxd/src/instructions/lsd/borrow_from_lsd_depository.rs

```diff
@@ -0,0 +1,284 @@
+
+    /// #
+    #[account(mut)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub liquidation_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub system_program: Program<'info, System>,
+
+    /// #
+    pub token_program: Program<'info, Token>,
+
+    /// #
+    pub uxd_program: Program<'info, crate::program::Uxd>,
+
+    /// #
+    pub rent: Sysvar<'info, Rent>,
+}
+
+pub(crate) fn handler(ctx: Context<BorrowFromLsdDepository>, collateral_amount: u64, loan_to_value_bps: u16) -> Result<()> {
+    // - 0 [CHECKS] ------------------------------------------------------------
+    {
+        // - 0.1 verify that borrows are enabled
+        let lsd_depository = ctx.accounts.depository.load()?;
+        require!(lsd_depository.borrowing_disabled == false, UxdError::BorrowingDisabled);
+
+    }
+
+    // - 1 [TRANSFER USER'S COLLATERAL] ----------------------------------------
+    {
+        msg!("[transfer] {} collateral from user to his lsd_position",
+            collateral_amount,
+        );
+        token::transfer(
+            ctx.accounts
+                .to_transfer_collateral_from_user_to_lsd_position_collateral_ta_context(),
+            collateral_amount,
+        )?;
+    }
+
+    // - 2 [MINT REDEEMABLES] --------------------------------------------------
+    let redeemable_borrow_amount = collateral_amount / u64::from(loan_to_value_bps) * u64::from(BPS_POWER);
+    {
+        msg!("[mint] {} redeemable to user (ltv {})",
+            collateral_amount,
+            u64::from(loan_to_value_bps) / BPS_POWER,
+        );
+        let controller_bump = ctx.accounts.controller.load()?.bump;
+        let controller_pda_signer: &[&[&[u8]]] = &[&[CONTROLLER_NAMESPACE, &[controller_bump]]];
+        token::mint_to(
+            ctx.accounts
+                .to_mint_redeemable_context()
+                .with_signer(controller_pda_signer),
+            redeemable_borrow_amount,
+        )?;
+    }
+
+    // - 3 [SETUP LIQUIDATION CRON] --------------------------------------------
+    let liquidation_price: u64;
+    {
+        let controller = &mut ctx.accounts.controller.load_mut()?;
+        let depository = &mut ctx.accounts.depository.load_mut()?;
+        let curtime = controller.get_time()?;
+        let collateral_oracle_params = depository.collateral_oracle_params;
+        let lsd_price = OraclePrice::new_from_oracle(
+            &ctx.accounts
+                .collateral_oracle_account
+                .to_account_info(),
+            &collateral_oracle_params,
+            curtime,
+            false,
+        )?;
+        let lsd_price_usd = lsd_price.get_asset_amount_usd(collateral_amount, depository.collateral_mint_decimals)?;
+        liquidation_price = maths::checked_mul(redeemable_borrow_amount, maths::checked_div(u64::from(depository.max_loan_to_value_bps), u64::from(BPS_POWER))?)?;
```

Comment:
sounds like we need a utility function to compute this, its used in many places

Reply:
Just 2 op I rather keep things flat x)
I know we diverge on this, but I think it's more straight forward. There might be a function in the math lib already though, else I would stick to what's there